staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -51,6 +51,9 @@ type CustomResourceDefinitionSpec struct {
 	// by GA > beta > alpha, and then by comparing major version, then minor version. An example sorted list of
 	// versions: v10, v2, v1, v11beta2, v10beta3, v3beta1, v12alpha1, v11alpha2, foo1, foo10.
 	Versions []CustomResourceDefinitionVersion `json:"versions,omitempty" protobuf:"bytes,7,rep,name=versions"`
+	// Prune enables pruning of unspecified fields. This required validation to be enabled.
+	// Note: this will default to true in version v1.
```

Comment:
In the next version.   Also, I think we should require this to enable versioning beyond the null conversion.